pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/service/phimanifest/PhiManifestService.java

```diff
@@ -0,0 +1,131 @@
+            return checkAnswerToActivity(participant, "CONSENT_ADDENDUM", "SOMATIC_CONSENT_ADDENDUM_TUMOR", true);
+        } else if (DBConstants.OSTEO_STUDY_GUID.equals(studyGuid)) {
+            return checkAnswerToActivity(participant, "CONSENT_ADDENDUM", "SOMATIC_CONSENT_TUMOR", true);
+        }
+        return false;
+    }
+
+    private boolean hasParticipantConsentedToSharedLearning(ElasticSearchParticipantDto participant) {
+        String dateOfBirth = participant.getDsm().get().getDateOfBirth();
+        int age = calculateAge(dateOfBirth);
+        if (age >= 7) {
+            return checkAnswerToActivity(participant, "CONSENT_ADDENDUM_PEDIATRIC", "SOMATIC_CONSENT_TUMOR_PEDIATRIC",
+                    true) && checkAnswerToActivity(participant, "CONSENT_ADDENDUM_PEDIATRIC",
+                    "SOMATIC_ASSENT_ADDENDUM", true);
+
+        } else if (age < 7) {
+            return checkAnswerToActivity(participant, "CONSENT_ADDENDUM_PEDIATRIC", "SOMATIC_CONSENT_TUMOR_PEDIATRIC",
+                    true);
+        }
+        return false;
+    }
+
+    /** checks if the answer provided by a participant to a survey question matches the criteria
+     * @param participant participant data from ES.
+     * */
+    private boolean checkAnswerToActivity(ElasticSearchParticipantDto participant, String activityName, String question, Object answer) {
+        Optional<Activities> maybeActivity = participant.getActivities().stream().filter(activities -> activities.getActivityCode()
+                .equals(activityName)).findAny();
+        if (maybeActivity.isPresent()) {
+            Activities activities = maybeActivity.get();
+            Optional<Map<String, Object>> maybeQuestionAnswers = activities.getQuestionsAnswers()
+                    .stream().filter(questionAnswer -> questionAnswer.get("stableId").equals(question)).findAny();
+            if (maybeQuestionAnswers.isPresent()) {
+                Map<String, Object> questionAnswer = maybeQuestionAnswers.get();
+                if (questionAnswer.containsKey("answer") && questionAnswer.get("answer").equals(answer)) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+
+    @VisibleForTesting
+    public boolean isSequencingOrderValid(@NonNull String sequencingOrderId, @NonNull String ddpParticipantId,
+                                          @NonNull DDPInstanceDto ddpInstanceDto) {
+        List<MercuryOrderDto> orders = new MercuryOrderDao().getByOrderId(sequencingOrderId);
+        if (orders.isEmpty() || orders.stream().anyMatch(order -> StringUtils.isBlank(order.getMercuryPdoId())
+                || ddpInstanceDto.getDdpInstanceId() != order.getDdpInstanceId()
+                || !ddpParticipantId.equals(order.getDdpParticipantId()))) {
+            return false;
+        }
+        return true;
+    }
+
+    public int calculateAge(@NonNull String dateOfBirth) {
```

Comment:
Consider moving to `DateTimeUtils` and changing the param to a stronger type, like date/localdate, and then calling the method something like ageInYears().